### PR TITLE
apps-vim-ide branch

### PR DESCRIPTION
**Pull request by MarkGreenPR**
This branch is a merge of the branches  jq-zsh-ctgas and vim-ide.
TODO: zshrc files, missing completion files, and vim plugin commentary

### CODE REVIEW

File: assets/templates/devkit/BOM_DEV_SOFTWARE.md [hunk 1]
```diff
@@ -1,18 +1,27 @@
 | Software       | Website                                   |
 |:---------------|:------------------------------------------|
-| bash           | https://www.gnu.org/software/bash/        |
+| bash           | https://www.gnu.org/software/bash         |
+| ctags          | http://ctags.sourceforge.net              |
+| direnv         | https://direnv.net                        |
 | docker         | https://www.docker.com                    |
-| docker-compose | https://docs.docker.com/compose/          |
+| docker-compose | https://docs.docker.com/compose           |
+| fossa-cli      | https://github.com/fossas/fossa-cli       |
```
Comment (jscharber) [inline assets/templates/devkit/BOM_DEV_SOFTWARE.md]: probably need to add fzf

I also install stern for tailing k8s logs and tilt as a front end to skaffold

File: assets/templates/devkit/BOM_SHELL_COMPS.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
+| Completion Package | Website |
```
Comment (jscharber) [inline assets/templates/devkit/BOM_SHELL_COMPS.md]: fzf and roadctl completion are also installed

File: devkit/BOM_DEV_SOFTWARE.md [hunk 3]
```diff
@@ -1,18 +1,27 @@
 | Software       | Website                                   |
 |:---------------|:------------------------------------------|
-| bash           | https://www.gnu.org/software/bash/        |
+| bash           | https://www.gnu.org/software/bash         |
+| ctags          | http://ctags.sourceforge.net              |
+| direnv         | https://direnv.net                        |
 | docker         | https://www.docker.com                    |
-| docker-compose | https://docs.docker.com/compose/          |
+| docker-compose | https://docs.docker.com/compose           |
+| fossa-cli      | https://github.com/fossas/fossa-cli       |
```
Comment (jscharber) [inline devkit/BOM_DEV_SOFTWARE.md]: fzf, stern, tilt, roadctl

File: devkit/BOM_SHELL_COMPS.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+| bash-completion    | https://github.com/scop/bash-completion |
+| zsh Completion     | https://github.com/zsh-users/zsh/tree/master/Completion |
+| zsh-completions    | https://github.com/zsh-users/zsh-completions |
+| go-zsh-completions | https://github.com/zchee/go-zsh-completions |
+
+| Command Completion | Website |
+|:-------------------|:--------|
+| docker        | https://github.com/docker/cli/tree/master/contrib/completion/bash |
+| docker-compose| https://github.com/docker/compose/tree/master/contrib/completion/bash |
+| git           | https://github.com/git/git/tree/master/contrib/completion/git-completion.bash |
```
Comment (jscharber) [inline devkit/BOM_SHELL_COMPS.md]: fzf, roadctl

File: devkit/states/git/pr_git_aliases [hunk 5]
```diff
@@ -0,0 +1,53 @@
+# vim: filetype=sh
+
+# Source from .bashrc or .zshrc
+
+alias gad="git add"
```
Comment (jscharber) [inline devkit/states/git/pr_git_aliases]: Do we want aliases for any other utilities?

File: devkit/states/golang/pr_go_aliases [hunk 6]
```diff
@@ -0,0 +1,53 @@
+# vim: filetype=sh
```
Comment (jscharber) [inline devkit/states/golang/pr_go_aliases]: I see more go aliases here

File: devkit/states/vim/pr_vimrc_init [hunk 7]
```diff
@@ -1,22 +1,364 @@
-" Set :GoDef and :GoInfo to use gopls
+" get all of the plugins
+source ~/.pr_vimrc_plug
+
+"-----------------------
+" airblade/vim-gitgutter
+"-----------------------
+" TODO disables default key maps why
```
Comment (jscharber) [inline devkit/states/vim/pr_vimrc_init]: I think it is safe to disable these in preference to using Fugitive

File: devkit/states/vim/pr_vimrc_init [hunk 8]
```diff
@@ -1,22 +1,364 @@
...
+let g:go_autodetect_gopath = 1
+
+" Default: set :GoTest timeout to 10 seconds
+let g:go_test_timeout = '10s'
+" Default: do not prepend name of test to error message
+let g:go_test_prepend_name = 1
+
+" Default: highlight all instances of identifier under cursor
+" let g:go_auto_sameids = 0
+" Default: do not show :GoInfo for word under cursor
```
Comment (jscharber) [inline devkit/states/vim/pr_vimrc_init]: I enable this by default, doesn't seem to cause a perf  problem

File: devkit/states/vim/pr_vimrc_init [hunk 9]
```diff
@@ -1,22 +1,364 @@
...
+noremap <S-F8> :NERDTreeToggle<CR>
+noremap <leader>nt :NERDTreeFind<CR>
+
+"-----------------
+" SirVer/ultisnips
+"-----------------
+let g:UltiSnipsExpandTrigger="<tab>"
+let g:UltiSnipsJumpForwardTrigger="<c-j>"
+let g:UltiSnipsJumpBackwardTrigger="<c-k>"
+" TODO the plugged directory is for plugins
```
Comment (jscharber) [inline devkit/states/vim/pr_vimrc_init]: There is a .vim/ultisnips directory but the snippets don't get installed there.  We can move them there.  Ultisnips using a different format of snippets, that is why I specified a directory that didn't include non-compatible snippets.

File: devkit/states/vim/pr_vimrc_init [hunk 10]
```diff
@@ -1,22 +1,364 @@
...
+
+" Default: set :GoTest timeout to 10 seconds
+let g:go_test_timeout = '10s'
+" Default: do not prepend name of test to error message
+let g:go_test_prepend_name = 1
+
+" Default: highlight all instances of identifier under cursor
+" let g:go_auto_sameids = 0
+" Default: do not show :GoInfo for word under cursor
+" TODO Enable if you do not find it too slow
```
Comment (jscharber) [inline devkit/states/vim/pr_vimrc_init]: On the short mappings.  I'm fine with your comments to change M to m, u to f, and y to k.

File: devkit/states/vim/pr_vimrc_init (hunk 9 above)
Comment (MarkGreenPR) [reply devkit/states/vim/pr_vimrc_init]: the default is ~/.vim/UltiSnips/vim.snippets, any reason not to leave it?

File: devkit/states/vim/pr_vimrc_init (hunk 10 above)
Comment (MarkGreenPR) [reply devkit/states/vim/pr_vimrc_init]: done

File: devkit/states/vim/pr_vimrc_init (hunk 8 above)
Comment (MarkGreenPR) [reply devkit/states/vim/pr_vimrc_init]: done

File: devkit/states/golang/pr_go_aliases (hunk 6 above)
Comment (MarkGreenPR) [reply devkit/states/golang/pr_go_aliases]: yes, added aliases for some go tools

File: devkit/states/vim/pr_vimrc_init (hunk 7 above)
Comment (MarkGreenPR) [reply devkit/states/vim/pr_vimrc_init]: done

File: assets/templates/devkit/BOM_DEV_SOFTWARE.md (hunk 1 above)
Comment (MarkGreenPR) [reply assets/templates/devkit/BOM_DEV_SOFTWARE.md]: done

Comment (MarkGreenPR) [reply assets/templates/devkit/BOM_DEV_SOFTWARE.md]: BOM fixed

File: devkit/states/git/pr_git_aliases (hunk 5 above)
Comment (MarkGreenPR) [reply devkit/states/git/pr_git_aliases]: possibly for commands that we install

File: devkit/BOM_DEV_SOFTWARE.md (hunk 3 above)
Comment (MarkGreenPR) [reply devkit/BOM_DEV_SOFTWARE.md]: added fzf and roadctl for now

File: devkit/BOM_SHELL_COMPS.md (hunk 4 above)
Comment (MarkGreenPR) [reply devkit/BOM_SHELL_COMPS.md]: also missing jq and ripgrep - done

**Comment by MarkGreenPR** [comment]
so far unresolved
1) add stern and tilt
2) add non go/git aliases
3) fix TODO on fzf in pr_vimrc_init
everything else was just pushed

File: devkit/BOM_DEV_SOFTWARE.md (hunk 3 above)
Comment (MarkGreenPR) [reply devkit/BOM_DEV_SOFTWARE.md]: added stern and tilt

**Pull request merged**